collect_app/src/test/java/org/odk/collect/android/preferences/ChangeAdminPasswordDialogTest.java

```diff
@@ -0,0 +1,166 @@
+import static junit.framework.TestCase.assertTrue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.robolectric.Shadows.shadowOf;
+
+@RunWith(RobolectricTestRunner.class)
+public class ChangeAdminPasswordDialogTest {
+
+    private ActivityController<FragmentActivity> activity;
+    private FragmentManager fragmentManager;
+    private ChangeAdminPasswordDialog dialogFragment;
+    private ChangeAdminPasswordDialog.ChangePasswordDialogCallback callback;
+
+    @Before
+    public void setup() {
+        activity = RobolectricHelpers.buildThemedActivity(FragmentActivity.class);
+        activity.setup();
+
+        fragmentManager = activity.get().getSupportFragmentManager();
+        dialogFragment = new ChangeAdminPasswordDialog();
+
+        callback = mock(ChangeAdminPasswordDialog.ChangePasswordDialogCallback.class);
+        dialogFragment.callback = callback;
+    }
+
+    @Test
+    public void dialogIsCancellable() {
+        dialogFragment.show(fragmentManager, "TAG");
+        assertThat(shadowOf(dialogFragment.getDialog()).isCancelable(), equalTo(true));
+    }
+
+    @Test
+    public void shouldShowCorrectButtons() {
+        dialogFragment.show(fragmentManager, "TAG");
+        AlertDialog dialog = (AlertDialog) ShadowDialog.getLatestDialog();
+
+        assertThat(dialog.getButton(DialogInterface.BUTTON_POSITIVE).getVisibility(), equalTo(VISIBLE));
+        assertThat(dialog.getButton(DialogInterface.BUTTON_POSITIVE).getText(),
+                equalTo(activity.get().getString(R.string.ok)));
+        assertThat(dialog.getButton(DialogInterface.BUTTON_NEGATIVE).getVisibility(), equalTo(VISIBLE));
+        assertThat(dialog.getButton(DialogInterface.BUTTON_NEGATIVE).getText(),
+                equalTo(activity.get().getString(R.string.cancel)));
+    }
+
+    @Test
+    public void clickingOkAfterSettingPassword_callsOnPasswordChanged() {
+        dialogFragment.show(fragmentManager, "TAG");
+        AlertDialog dialog = (AlertDialog) ShadowDialog.getLatestDialog();
+        EditText passwordEditText = dialog.findViewById(R.id.pwd_field);
+        passwordEditText.setText("blah");
+        dialog.getButton(DialogInterface.BUTTON_POSITIVE).performClick();
+
+        verify(callback, times(1)).onPasswordChanged("blah");
+    }
+
+    @Test
+    public void passwordIsRetainedOnScreenRotation() {
+        dialogFragment.show(fragmentManager, "TAG");
+        AlertDialog dialog = (AlertDialog) ShadowDialog.getLatestDialog();
+        EditText passwordEditText = dialog.findViewById(R.id.pwd_field);
+        passwordEditText.setText("blah");
+
+        assertThat(activity.get().getResources().getConfiguration().orientation, equalTo(Configuration.ORIENTATION_PORTRAIT));
```

Comment:
This isn't really the correct logic for what happens when an Activity rotates. Normally and Activity will go through a full onPause/onStop/onDestroy run and then go back through the creation steps. As part of this the Fragment will be restored (using the arguments it had before rotation). Instead of trying to deal with all this yourself you should be able to the ActivityScenario like in `AdminPasswordDialogFragmentTest#afterRecreating_enteringPassword_andClickingOK_callsOnCorrectAdminPasswordWithAction`.